### DISCUSSION TITLE
Updating Current System

### DISCUSSION THREAD

**Discussion by jeremykenedy** [serversideup/financial-freedom#59] [General]
Hi,

I have updated the project to Laravel 10, and systematically gone through every single file and dependency. I have also updated the build to use vitejs.

I am still working out some kinks in the running it hot and then need to resolve an error with the user model appends causing a blocking render error.

I have also added prettier and eslint. Eslint reveals tons of errors in which I have suppressed within the config leaving major errors to be resolved in code. You can run `yarn run lint` to show them.

I will make a PR once I have worked out the kinks but wanted to give you a heads up as this a major core overhaul of what is existing to be updated to the most recent version of laravel, composer packages, and npm packages.

If I make a PR will you get to it sooner than later?

Thanks,

Jeremy

The WIP can be seen here:
https://github.com/jeremykenedy/financial-freedom
https://github.com/jeremykenedy/financial-freedom/commit/1835700c8639f4829ea3607c9fe0f96776f99e22

**Comment by jaydrogers** [comment]
Hey Jeremy!

Thanks for your continued interest in the project! We're grateful for your eagerness to jump in an help out.

Just to explain a little more where we expect Financial Freedom once @danpastori and I have time to work on it again:

**We expect a lot of the current features to receive a major overhaul**
Upgrading the technical side isn't the time consuming part for us. The major time consuming part is refocusing and looking at this project not as a side project any more. This means organizing features, creating contribution guides, and being able to communicate our vision with the open source community.

**This probably means a major refactor is in the future**
We will start from the drawing board again and bring a whole new perspective on it. This likely means that a lot of code will change.

**Next steps**
Unfortunately, the best steps for now is to let @danpastori and I get through our work right now. When we come back to this project, we have a ton new ideas, features, and experience that we will be applying to this project.

If you'd like to fork what we currently have and work on it for yourself -- by all means. But for the next version of Financial Freedom, expect this to be a strong and stable response to the accidental attention this project received 😃

Looking forward to getting to this soon. Trust me, we're just as anxious to get to this!

**Reply by jeremykenedy** [reply]
This doesn't change any functionality but is a core update to the system so you can use it if you want but based on that I'm not going to waste any more time into it, that's why I probed via an issue.  If you need any further help, my hourly rate is $140 and require an NDA, and contract.

**Reply by jaydrogers** [reply]
Thanks @jeremykenedy! @danpastori will review and will keep you posted of any questions.

Can't wait to have the new version of Financial Freedom out soon 😃🏃‍♂️

**Reply by jeremykenedy** [reply]
I'm down to contribute for free though as I enjoy open source work. I do believe in high quality work and design patterns. 😉

**Reply by jaydrogers** [reply]
Thanks! Stay tuned. We have a ton of things planned for it 🚀

**Reply by jaydrogers** [reply]
Thanks for your sponsorship @jeremykenedy! If you join our Discord, we can get you labeled as a sponsor too. Your avatar should appear automatically soon 😃

https://serversideup.net/discord
